### PR TITLE
Firbaseauth missingimplementation

### PR DESCRIPTION
**Pull request by vuppalasivakumar**
HI I had Implemented phone number authentication and delete user

### CODE REVIEW

File: packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java [hunk 1]
```diff
@@ -96,12 +111,76 @@ public void onMethodCall(MethodCall call, Result result) {
       case "stopListeningAuthState":
         handleStopListeningAuthState(call, result);
         break;
+        case "deleteCurrentUser":
+            handledeletecurrentuser(result);
```
Comment (jonasbark) [inline packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java]: method names should be camel cased

File: packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java [hunk 2]
```diff
@@ -96,12 +111,76 @@ public void onMethodCall(MethodCall call, Result result) {
...
+      @Override
+      public void onVerificationCompleted(PhoneAuthCredential phoneAuthCredential) {
+
+
+        firebaseAuth.signInWithCredential(phoneAuthCredential)
+                .addOnCompleteListener(activity, new SignInCompleteListenerforphoneauth(result));
+      }
+      @Override
+      public void onVerificationFailed(FirebaseException e) {
+        Log.e("onVerificationFailed", e.toString());
```
Comment (jonasbark) [inline packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java]: debug log entries should be removed I guess

File: packages/firebase_auth/example/android/app/google-services.json [hunk 3]
```diff
@@ -1,27 +1,58 @@
 {
   "project_info": {
-    "project_number": "297855924061",
-    "firebase_url": "https://flutterfire-cd2f7.firebaseio.com",
-    "project_id": "flutterfire-cd2f7",
-    "storage_bucket": "flutterfire-cd2f7.appspot.com"
+    "project_number": "931992176420",
```
Comment (jonasbark) [inline packages/firebase_auth/example/android/app/google-services.json]: you don't want to have your private information here

File: packages/firebase_auth/example/ios/Runner/AppDelegate.m [hunk 4]
```diff
@@ -4,13 +4,110 @@
...
+        UNAuthorizationOptionAlert
+        | UNAuthorizationOptionSound
+        | UNAuthorizationOptionBadge;
+        [[UNUserNotificationCenter currentNotificationCenter] requestAuthorizationWithOptions:authOptions completionHandler:^(BOOL granted, NSError * _Nullable error) {
+        }];
+#endif
+    }
+    
+    [application registerForRemoteNotifications];
+    [FIRMessaging messaging].delegate = self;
```
Comment (jonasbark) [inline packages/firebase_auth/example/ios/Runner/AppDelegate.m]: the Firebase Messaging stuff shouldn't be in this commit

File: packages/firebase_auth/example/ios/Runner/Info.plist [hunk 5]
```diff
@@ -18,6 +18,17 @@
 	<string>1.0</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
+	<key>CFBundleURLTypes</key>
```
Comment (jonasbark) [inline packages/firebase_auth/example/ios/Runner/Info.plist]: that stuff as well

File: packages/firebase_auth/example/ios/Runner/AppDelegate.m (hunk 4 above)
Comment (Prayuta) [reply packages/firebase_auth/example/ios/Runner/AppDelegate.m]: This is required for using the feature of swizzling notification which is part of the phone number authentication in the iOS Variant.

**Comment by Hixie** [comment]
cc @collinjackson

**Comment by collinjackson** [comment]
This CL touches some files in other plugins, which probably wasn't your intent. It looks like the phone auth functionality is also being added in #329, so I'm closing this one, but please let me know if you'd like me to reopen it.

I do think we do want the delete user functionality... if you'd like to continue working on that, please open a new pull request. Thanks!

**Pull request closed without merging**